client/web/web.go

```diff
@@ -202,6 +206,23 @@ func (s *Server) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 }
 
 func (s *Server) serve(w http.ResponseWriter, r *http.Request) {
+	// serve HTTP 204 on /ok requests as connectivity check
+	if r.Method == httpm.GET && r.URL.Path == "/ok" {
+		w.WriteHeader(http.StatusNoContent)
+		return
+	}
+
+	if s.mode == ManageServerMode {
+		// In manage mode, requests must be sent directly to the bare Tailscale IP address.
+		// If a request comes in on any other hostname, redirect.
+		if s.requireTailscaleIP(w, r) {
+			return
+		}
+		w.Header().Set("X-Frame-Options", "DENY")
+		w.Header().Set("Content-Security-Policy", "default-src 'self'")
+		w.Header().Set("Cross-Origin-Resource-Policy", "same-origin")
```

Comment:
Should we also set any of these for "login" mode?

Reply:
some of them, possibly yes.  I just hadn't given login mode quite as much thought just yet, and was instead focused on locking down the management client.

We definitely wouldn't want this x-frame-options header for login, since on unraid and home assistant, the client is initially rendered in an iframe.  Maybe a different value for that header, or none at all?  We also need to allow cross-site requests from the lobby client to the full client in order to detect connectivity.  That will require relaxing some headers.